#167377554 Search functionality for articles

Nennyfills opened the pull request:
### What does this PR do?
This pull request enables all user to search for articles.

#### Description of Task to be completed
-  Create RESTful route to get an article by category, title, author, and keyword
-  Add validation to articles category, title, author, and keyword queries
-  Write a test to test articles validation and route

#### How should this be manually tested?
- git clone this branch
- run `npm install`
- Using postman: 
    Get  - `http://localhost:{port}/api/v1/articles?author`

#### Any Background Context You want to provide?
N/A
#### What are the relevant pivotal tracker stories?

[#167377554](https://pivotaltracker.com/story/show/167377554)

#### Screenshots (if appropriate)
<img width="1116" alt="Screenshot 2019-08-07 at 1 29 53 PM" src="https://user-images.githubusercontent.com/45515619/62624979-39b69280-b91c-11e9-94ad-ba823ff4ce9c.png">
<img width="1023" alt="Screenshot 2019-08-07 at 1 28 42 PM" src="https://user-images.githubusercontent.com/45515619/62624986-3d4a1980-b91c-11e9-9466-2484bb4fe532.png">
<img width="1080" alt="Screenshot 2019-08-07 at 1 28 13 PM" src="https://user-images.githubusercontent.com/45515619/62624996-41763700-b91c-11e9-9cd9-0ef3f3453e79.png">
<img width="1084" alt="Screenshot 2019-08-07 at 1 27 44 PM" src="https://user-images.githubusercontent.com/45515619/62625005-44712780-b91c-11e9-9531-508e1dc0d176.png">
<img width="1440" alt="Screenshot 2019-08-07 at 1 27 31 PM" src="https://user-images.githubusercontent.com/45515619/62625019-4c30cc00-b91c-11e9-81a5-f8b11e7f7176.png">
<img width="541" alt="Screenshot 2019-08-07 at 1 26 55 PM" src="https://user-images.githubusercontent.com/45515619/62625027-505ce980-b91c-11e9-9d3e-0c094818f1aa.png">

#### Questions:
N/A

Merkll commented on src/helpers/genericArticleHelper.js:
```diff
@@ -0,0 +1,103 @@
+    const suspended = entry['article.suspended'];
+    const status = entry['article.status'];
+    const coverImageUrl = entry['article.coverImageUrl'];
+    const followUpId = entry['article.title'];
```
let's use ```Array.map``` in place of ```forEach```

Nennyfills replied:
Done.

Merkll commented on src/helpers/paginator.js:
```diff
@@ -7,6 +7,7 @@ import { NotFoundError } from './errors';
  *
  * @param {Object} Source - Model object
  * @param {Object} options
+//  * @param {query} query - For model query
```
Check and Remove all comments

Nennyfills replied:
Done.

Nennyfills replied:
Done

Merkll commented on src/helpers/genericArticleHelper.js:
```diff
@@ -0,0 +1,90 @@
+    author: {
+      [Op.or]: {
+        '$author.firstName$': { [Op.iLike]: `%${author}%` },
+        '$author.lastName$': { [Op.iLike]: `%${author}%` },
```
Good Job searching in all fields

Nennyfills replied:
Thanks

Nennyfills replied:
Thanks.

Nennyfills committed:
feat(search): search functionality for articles

- create RESTful route to get article by tag, title, author and keyword
- add validalition to articles tag, title, author and keyword queries
-write test to test articles validalition and route

[Finishes 167377554]
